### PR TITLE
Privacy configuration tests

### PR DESCRIPTION
**Pull request by kdzwinel**
NOTE - might be worth merging https://github.com/duckduckgo/privacy-reference-tests/pull/10 first to make sure that this PR passes linting (I tested locally but you never know).

Implementation guidelines for the feature are here: https://app.asana.com/0/1198207348643509/1200553491728590/f

Test implementation in the web extension is waiting here: https://github.com/duckduckgo/duckduckgo-privacy-extension/compare/konrad/reference-tests-config

This follows the new format outlined here: https://app.asana.com/0/0/1201321554045382/f

### CODE REVIEW

File: privacy-configuration/config1_reference.json [hunk 1]
```diff
@@ -0,0 +1,167 @@
+{
```
Comment (kdzwinel) [inline privacy-configuration/config1_reference.json:1]: config files are pretty unexciting. TLDR: config1 just has all features disabled, config2 has all features enabled but also has a bunch of exceptions, config3 is an edge-case where feature key is missing

File: privacy-configuration/tests.json [hunk 2]
```diff
@@ -0,0 +1,630 @@
+{
+    "featuresDisabled": {
+        "name": "All features are disabled",
+        "desc": "Those tests use config1_reference.json where all features are disabled",
+        "referenceConfig": "config1_reference.json",
```
Comment (kdzwinel) [inline privacy-configuration/tests.json:5]: this is new - I wanted to make it automatable to pair each test set with the right config

File: privacy-configuration/tests.json [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+                "name": "click to load should be disabled",
+                "featureName": "clickToPlay",
+                "siteURL": "https://example.org",
+                "expectFeatureEnabled": false,
+                "exceptPlatforms": [
+                    "safari-extension",
+                    "ios-browser",
+                    "android-browser",
+                    "macos-browser",
+                    "windows-browser"
```
Comment (kdzwinel) [inline privacy-configuration/tests.json:63]: I came up with names for all of the platforms (I think they were not defined anywhere) based on the existing format (we use "ios-browser" in other set of tests)

**Review by SlayterDev — APPROVED** [review]
LGTM 👍

**Review by ladamski — APPROVED** [review]
For the first config file, is there value in having all of the duplicate tests towards the end (eg. from "clickToPlay" onward)?   If they're completely redundant maybe better to skip them entirely?  
Otherwise LGTM!

**Comment by kdzwinel** [comment]
@ladamski thanks for reviewing!

> , is there value in having all of the duplicate tests towards the end (eg. from "clickToPlay" onward)?

Not sure what you mean exactly. Do you mean the `featuresDisabled` set? Yeah, it's a pretty boring test set and `featuresEnabled` is no different. I didn't know if it makes sense to check all features this way, but I guess this depends on a implementation e.g. in the extension we check if all features are enabled in the exact same way `loadConfig(config); isFeatureEnabled(featureName)`, but on other platforms each feature can have a separate `isEnabled` implementation. For other test sets (e.g. `localExceptions`) I decided against testing every feature. My thinking was that featuresDisabled/featuresEnabled verify that feature is even "plugged into" config, but for rest of the tests I'm not re-testing if e.g. exceptions work the same way for every feature. Happy to discuss more, I'm not attached to my approach.

**Comment by ladamski** [comment]
> Not sure what you mean exactly. Do you mean the `featuresDisabled` set? Yeah, it's a pretty boring test set and `featuresEnabled` is no different. I didn't know if it makes sense to check all features this way, but I guess this depends on a implementation e.g. in the extension we check if all features are enabled in the exact same way `loadConfig(config); isFeatureEnabled(featureName)`, but on other platforms each feature can have a separate `isEnabled` implementation. For other test sets (e.g. `localExceptions`) I decided against testing every feature. My thinking was that featuresDisabled/featuresEnabled verify that feature is even "plugged into" config, but for rest of the tests I'm not re-testing if e.g. exceptions work the same way for every feature. Happy to discuss more, I'm not attached to my approach.

Ok that makes sense, yes I was wondering if it made sense to test each disabled feature, but if there's a chance some platforms implement those then it makes sense to have them all there.
